src/ripple/app/hook/impl/applyHook.cpp

```diff
@@ -4609,6 +4613,116 @@ DEFINE_HOOK_FUNCTION(
     HOOK_TEARDOWN();
 }
 
+DEFINE_HOOK_FUNCTION(
+    int64_t,
+    util_verify_p256,
+    uint32_t hread_ptr,  // hash
+    uint32_t hread_len,
+    uint32_t rread_ptr,  // r of signature
+    uint32_t rread_len,
+    uint32_t sread_ptr,  // s of signature
+    uint32_t sread_len,
+    uint32_t xread_ptr,  // x of public key
+    uint32_t xread_len,
+    uint32_t yread_ptr,  // y of public key
+    uint32_t yread_len)
+{
+    HOOK_SETUP();  // populates memory_ctx, memory, memory_length, applyCtx,
+                   // hookCtx on current stack
+
+    if (NOT_IN_BOUNDS(hread_ptr, hread_len, memory_length) ||
+        NOT_IN_BOUNDS(rread_ptr, rread_len, memory_length) ||
+        NOT_IN_BOUNDS(sread_ptr, sread_len, memory_length) ||
+        NOT_IN_BOUNDS(xread_ptr, xread_len, memory_length) ||
+        NOT_IN_BOUNDS(yread_ptr, yread_len, memory_length))
+        return OUT_OF_BOUNDS;
+
+    if (hread_len != 32)
+        return INVALID_ARGUMENT;
+
+    if (rread_len > 32 || sread_len > 32 || xread_len > 32 || yread_len > 32)
+        return TOO_BIG;
+
+    // create curve object
+    EC_GROUP* group = EC_GROUP_new_by_curve_name(NID_X9_62_prime256v1);
+    if (!group)
```

Comment:
why p256 and not secp256k1/ed25591 ?

Reply:
Verification for secp256k1/ed25591 is already available in util_verify.
The reason for adding P256 verification here is to support passkey.